AsyncDisplayKit/Layout/ASLayout.mm

```diff
@@ -68,6 +70,7 @@ - (instancetype)initWithLayoutElement:(id<ASLayoutElement>)layoutElement
 #endif
     
     _layoutElement = layoutElement;
+    _layoutElementType = layoutElement.layoutElementType;
```

Comment:
Nit but maybe we could add a comment why we do that in this case. Otherwise in the future maybe we will look over it and wonder why we not reading it directly in the `type` method